File: test/ayush.dev/python/pipelines/PythonInsideProjectDatasetsOnly/code/pythoninsideprojectdatasetsonly/graph/FlattenSchema_1.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from pythoninsideprojectdatasetsonly.config.ConfigStore import *
from pythoninsideprojectdatasetsonly.udfs.UDFs import *

def FlattenSchema_1(spark: SparkSession, in0: DataFrame) -> DataFrame:
    return in0\
        .withColumn("c_array_decimal", explode_outer("c_array_decimal"))\
        .withColumn("c_array_float", explode_outer("c_array_float"))\
        .withColumn("c_struct-c_array_int", explode_outer("c_struct.c_array_int"))\
        .select(col("c_array_float"), col("c_array_decimal"), col("c_struct.c_array_int").alias("c_struct-c_array_int"))

File: test/ayush.dev/scala/pipelines/ScalaInsideProjectDatasetsOnly/code/src/main/scala/io/prophecy/pipelines/scalainsideprojectdatasetsonly/graph/FlattenSchema_1.scala
package io.prophecy.pipelines.scalainsideprojectdatasetsonly.graph

import io.prophecy.libs._
import io.prophecy.pipelines.scalainsideprojectdatasetsonly.config.ConfigStore._
import io.prophecy.pipelines.scalainsideprojectdatasetsonly.udfs.UDFs._
import io.prophecy.pipelines.scalainsideprojectdatasetsonly.udfs._
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object FlattenSchema_1 {

  def apply(spark: SparkSession, in: DataFrame): DataFrame =
    in.withColumn("c_array_decimal",   explode_outer(col("c_array_decimal")))
      .withColumn("c_array_timestamp", explode_outer(col("c_array_timestamp")))
      .withColumn("c_struct-c_array_int",
                  explode_outer(col("c_struct.c_array_int"))
      )
      .select(col("c_array_decimal"),
              col("c_array_timestamp"),
              col("c_struct.c_array_int").as("c_struct-c_array_int")
      )

}
